### ISSUE TITLE
TEP-0142 Promote StepActions to Beta

### ISSUE THREAD

**Issue by chitrangpatel** [tektoncd/pipeline#7805]
This issue tracks the migration of StepActions, targeting [v0.60.0](https://github.com/tektoncd/pipeline/releases/tag/v0.60.0).

https://github.com/tektoncd/pipeline/issues/7424 describes the feedback and feature requests we have been getting. All the blocking work seems to be almost completed. What's remaining are nice-to-have features which can be implemented in parallel. There are no issues related to the StepActions API so I think it can be promoted to Beta by [v0.60.0](https://github.com/tektoncd/pipeline/releases/tag/v0.60.0).

### Feature looking for Promotion

**StepActions**

https://github.com/tektoncd/community/blob/main/teps/0142-enable-step-reusability.md

Docs: 
- https://github.com/tektoncd/pipeline/blob/main/docs/stepactions.md

### Current Feature Flag

`enable-step-actions: false`

### Maintain Feature Flag as opt-in when promoting to Beta

`enable-step-actions: false`

### Pipeline Release

[v0.54.0](https://github.com/tektoncd/pipeline/releases/tag/v0.54.0)

### Issues Reported by the Users

- https://github.com/tektoncd/pipeline/issues/7754
- https://github.com/tektoncd/pipeline/issues/7764

### Dogfooding

Yes, https://github.com/tektoncd/plumbing/pull/1849

### Catalog
- [x] Git: https://github.com/tektoncd/catalog/tree/main/stepaction
- [x] Tekton Bundles: gcr.io/tekton-releases/catalog/upstream/stepactions
- [x] ArtifactHub: https://github.com/artifacthub/hub/issues/3677#issuecomment-1987754079

### Examples

- https://github.com/tektoncd/pipeline/blob/main/examples/v1/taskruns/alpha/stepaction-git-resolver.yaml
- https://github.com/tektoncd/pipeline/blob/main/examples/v1/taskruns/alpha/stepaction-params.yaml
- https://github.com/tektoncd/pipeline/blob/main/examples/v1/taskruns/alpha/stepaction-passing-results.yaml
- https://github.com/tektoncd/pipeline/blob/main/examples/v1/taskruns/alpha/stepaction-results.yaml
- https://github.com/tektoncd/pipeline/blob/main/examples/v1/taskruns/alpha/stepaction.yaml

/kind feature

**Comment by chitrangpatel** [comment]
Thanks @ericzzzzzzz for your work on this!!

/meow
